Cython/Compiler/ExprNodes.py

```diff
@@ -10154,57 +10155,52 @@ def may_be_none(self):
 
     def calculate_result_code(self, code=None):
         if self.result_code is None:
-            self.result_code = code.get_py_const(py_object_type, 'codeobj', cleanup_level=2)
+            self.result_code = code.get_py_codeobj_const(self)
         return self.result_code
 
     def generate_result_code(self, code):
         if self.result_code is None:
-            self.result_code = code.get_py_const(py_object_type, 'codeobj', cleanup_level=2)
+            # cname already generated
+            self.result_code = code.get_py_codeobj_const(self)
 
-        code = code.get_cached_constants_writer(self.result_code)
-        if code is None:
-            return  # already initialised
-        code.mark_pos(self.pos)
         func = self.def_node
-        func_name = code.get_py_string_const(
+        self.func_name_result = code.get_py_string_const(
             func.name, identifier=True, is_str=False, unicode_value=func.name)
         # FIXME: better way to get the module file path at module init time? Encoding to use?
         file_path = StringEncoding.bytes_literal(func.pos[0].get_filenametable_entry().encode('utf8'), 'utf8')
-        file_path_const = code.get_py_string_const(file_path, identifier=False, is_str=True)
+        self.file_path_result = code.get_py_string_const(file_path, identifier=False, is_str=True)
+
+        self.varnames_result = self.varnames.result()
```

Comment:
Can't the values of these attributes also be created locally in `generate_codeobj_tab_entry`? They don't seems to be special in any way, or dependent on any local state in this method.

Reply:
It's more than I think it'd probably be too late to generate them in `generate_codeobj_tab_entry` because we've probably already written the stringtab by then. I'll add a comment though because it isn't obvious

Reply:
Not sure. The string tab has its own position in the code sections and even utility code that comes in during code generation can still add strings to it. Have you tried it?

Reply:
I tried moving the string tab and integer constant generation calls last in `generate_const_declarations` and it still worked for me. That's probably a good idea anyway, the simpler the constants, the more chances we should give move elaborate code to make use of them.

```patch
diff --git a/Cython/Compiler/Code.py b/Cython/Compiler/Code.py
index eb41b8065..37ed1e61c 100644
--- a/Cython/Compiler/Code.py
+++ b/Cython/Compiler/Code.py
@@ -1523,9 +1523,9 @@ class GlobalState:
 
     def generate_const_declarations(self):
         self.generate_cached_methods_decls()
+        self.generate_object_constant_decls()
         self.generate_string_constants()
         self.generate_num_constants()
-        self.generate_object_constant_decls()
 
     def generate_object_constant_decls(self):
         consts = [(len(c.cname), c.cname, c)
```